main.py

```diff
@@ -544,19 +618,39 @@ def _send_flow_mod(self, switch, flow_mod):
 
     def _send_napp_event(self, switch, flow, command, **kwargs):
         """Send an Event to other apps informing about a FlowMod."""
-        if command == "add":
-            name = "kytos/flow_manager.flow.added"
-        elif command in ("delete", "delete_strict"):
-            name = "kytos/flow_manager.flow.removed"
-        elif command == "error":
-            name = "kytos/flow_manager.flow.error"
-        else:
-            raise InvalidCommandError
+        command_events = {
+            "pending": "kytos/flow_manager.flow.pending",
+            "add": "kytos/flow_manager.flow.added",
+            "delete": "kytos/flow_manager.flow.removed",
+            "delete_strict": "kytos/flow_manager.flow.removed",
+            "error": "kytos/flow_manager.flow.error",
+        }
+        try:
+            name = command_events[command]
+        except KeyError as error:
+            raise InvalidCommandError(str(error))
         content = {"datapath": switch, "flow": flow}
         content.update(kwargs)
         event_app = KytosEvent(name, content)
         self.controller.buffers.app.put(event_app)
 
+    @listen_to("kytos/core.openflow.connection.error")
+    def on_openflow_connection_error(self, event):
+        """Listen to openflow connection error and publish the flow error."""
+        switch = event.content["destination"].switch
+        flow = event.message
+        try:
+            _, error_command = self._flow_mods_sent[event.message.header.xid]
+        except KeyError:
+            error_command = "unknown"
+        self._send_napp_event(
+            switch,
+            flow,
+            "error",
+            error_command=error_command,
+            error_exception=event.content.get("exception"),
```

Comment:
@ajoaoff @italovalcy we might need to augment [`mef_eline`'s method `handle_flow_mod_error`](https://github.com/kytos-ng/mef_eline/blob/master/main.py#L668-L674) - if we keep using the `'kytos/flow_manager.flow.error'` as we're using here to also bubble up the socket error of a flow mod install, in case the `error_exception` is set, it should be treated the same way as a `SwitchNotConnectedError` that results in an HTTP 424, because when this happens even though the consistency check will try to add or delete later on the client might want to try another path when applicable (we don't know for how long this might keep happening), and if clients don't handle this, then traffic could be lost because temporarily until the consistency check send again, network traffic might be using other flow entries that are part of the circuit. Let me know your preference on how to handle this. I haven't mapped an issue on `mef_eline` for this yet, let's evolve the discussion.

Reply:
@viniarck if I understood correctly, upon receiving a `kytos/flow_manager.flow.error` and  `error_command=='add'` (the original request was to add a flow), mef_eline will call `evc.remove_current_flows()` (https://github.com/kytos-ng/mef_eline/blob/master/main.py#L690) which is basically the same way it handles the `SwitchNotConnectedError` that results in an HTTP 424 (https://github.com/kytos-ng/mef_eline/blob/master/models/evc.py#L514). Then, after removing the currently flows, the EVC will be marked as `active == False` (and theoretically it is `enabled == True`), leaving to the consistency check reactivate it (https://github.com/kytos-ng/mef_eline/blob/master/main.py#L66-L70). Please let me know if you meant something different.

Reply:
Right, @italovalcy, yeah, this current error handling approach on `mef_eline` should work as intended. 

Originally, I meant to bring the discussion to this point here when the `error_exception` is also set (in addition to having the `error_command=='add'`, maybe this would just be a temporary Socket failure, so it could be re-tried a few times or even try to use another path, **but as we currently have it's save to try to remove since `mef_eline` will use the newly iplemented `force` option as well**. 

Thinking about an enhancement though, If we need to try to find an alternative path we could try to enhance this part here. It comes down to try to find an alternative path, if any, or keep retrying, when this type of exception is raised. Or, arguably, to facilitate for clients, maybe on `flow_manager` we could also have some retries there for such cases when the `error_exception` happened, and then only raise this error after a few attempts, and then once clients got this event with the exception, it's safe to assume that retries aren't worth anymore, and either they try to find an alternative path or completely remove. Wdyt?

Reply:
Understood @viniarck, that is a good point! Thanks for bringing this enhancement possibility. IMHO, I think it is a good idea to have the flow_manager trying a few more times before reporting the error to the requestor. I support this idea. Tow points to take into consideration for this proposed approach: 1) how can we differentiate permanent and temporary errors? 2) what interval should we use between each try?

The reason for 1 is: most of the times I've seen errors like this is due to a switch reconnection (temporary error). Switch reconnection (during the installation of a flow) can be a good case where the retries approach fits better than just removing the current path and waiting for the next consistency check. In the order hand, permanent errors such as invalid action, invalid port, table full, etc. those are not worth to keep trying (if possible to differentiate those).

For 2, maybe we can reply an exponencial backoff approach starting with a few milliseconds. Example: wait 100ms and try again, wait 200ms and try again, finally wait 400ms and try again, give up and generate the `kytos/flow_manager.flow.error `.

Reply:
Yeah, for the point number 1) `kytos/core.openflow.connection.error` events are potential temporary errors, and ultimately, as we currently have here, it'll send a `kytos/flow_manager.flow.error` and set the `error_exception`, that way subscribers can still different them from `OFPT_ERROR` which will also be sent as `kytos/flow_manager.flow.error` but without the `error_exception` key, and then subscribers can take actions accordingly as needed. 

But, maybe having a new event on flow_manager `kytos/flow_manager.connection.error` (this one being a higher level version of the `kytos/core.openflow.connection.error` with the retries, that way also not mixing contexts with the `kytos/flow_manager.flow.error` that should be exclusive for `OFPT_ERROR`s as it was meant to be originally) would probably make things easier to follow and understand and for subscribers to handle and differentiate, wdyt? 

For 2), yep, for sure, I think exponential short retries should fit this pretty well, and it wouldn't impact or add too much delay, allowing subscribers that need responsiveness to still have it. I appreciated your feedback and suggestions @italovalcy, I'll map this in an issue.

Reply:
Thanks for the comments, @viniarck. Yeah, having the retries approach configurable is the best way forward. Regarding the discussion about a new event on flow_manager, I rather prefer the first suggestion you gave: clients listening to `kytos/flow_manager.flow.error` and differentiating temp/permanent errors by looking at the `error_exception` (or even don't send a `kytos/flow_manager.flow.error` before running out of retries).

Reply:
Sounds good @italovalcy, I've captured this requirement and refined info that we've discussed here https://github.com/kytos-ng/flow_manager/issues/56. thanks for you review and feedback.